hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterFileSystem.java

```diff
@@ -221,7 +221,7 @@ public Path getWALRootDir() {
    * @return the directory for a give {@code region}.
    */
   public Path getRegionDir(RegionInfo region) {
-    return FSUtils.getRegionDir(FSUtils.getTableDir(getRootDir(), region.getTable()), region);
+    return FSUtils.getRegionDirFromRootDir(getRootDir(), region);
```

Comment:
Why this? The old impl should be getRegionDirFromTableDir?

Reply:
They are the same...

You can see the implementation of getRegionDirFromRootDir, it will first generate the tableDir...